craft_providers/lxd/lxc.py

```diff
@@ -0,0 +1,972 @@
+
+        :raises LXDError: on unexpected error.
+        """
+        command = ["config", "set", f"{remote}:{instance_name}", key, value]
+
+        try:
+            self._run_lxc(
+                command,
+                capture_output=True,
+                check=True,
+                project=project,
+            )
+        except subprocess.CalledProcessError as error:
+            raise LXDError(
+                brief=f"Failed to set config key {key!r} to {value!r} for instance {instance_name!r}.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+    def delete(
+        self,
+        *,
+        instance_name: str,
+        force=False,
+        project: str = "default",
+        remote: str = "local",
+    ) -> None:
+        """Delete instance.
+
+        :param instance_name: Name of instance.
+        :param force: Force deletion if running.
+        :param project: Name of LXD project.
+        :param remote: Name of LXD remote.
+
+        :raises LXDError: on unexpected error.
+        """
+        command = ["delete", f"{remote}:{instance_name}"]
+
+        if force:
+            command.append("--force")
+
+        try:
+            self._run_lxc(
+                command,
+                capture_output=True,
+                check=True,
+                project=project,
+            )
+        except subprocess.CalledProcessError as error:
+            raise LXDError(
+                brief=f"Failed to delete instance {instance_name!r}.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+    def exec(
+        self,
+        *,
+        command: List[str],
+        instance_name: str,
+        cwd: str = "/root",
+        mode: str = "auto",
+        project: str = "default",
+        remote: str = "local",
+        runner=subprocess.run,
+        **kwargs,
+    ):
+        """Execute command in instance_name with specified runner.
+
+        :param command: Command to execute in the instance.
+        :param instance_name: Name of instance to execute in.
+        :param cwd: Current working directory for command.
+        :param mode: Override terminal mode Valid options include: "auto",
+            "interactive", "non-interactive":w
+        :param project: Name of LXD project.
+        :param remote: Name of LXD remote.
+        :param runner: Execution function to invoke, e.g. subprocess.run or
+            Popen.  First argument is finalized command with the attached
+            kwargs.
+        :param kwargs: Additional kwargs for runner.
+
+        :returns: Runner's instance.
+        """
+        final_cmd = [
+            str(self.lxc_path),
+            "--project",
+            project,
+            "exec",
+            f"{remote}:{instance_name}",
+        ]
+
+        if cwd != "/root":
+            final_cmd.extend(["--cwd", cwd])
+
+        if mode != "auto":
```

Comment:
```suggestion
        if mode is not None:
```

Reply:
Wondering if this should be just `if mode`, for the case where `mode` is an empty string.